Branch Types Implementation

malcolmriley opened the pull request:
Closes #63, closes #64.

Implements different functionality for various branch types.
- **Breaky Branch**: After landing on this branch 3 times, it breaks.
- **Bouncy Branch**: This branch launches you upwards.
- **Slippery Branch**: This branch imparts a random horizontal velocity on the player when they land on it.
- **Normal Branch**: This branch doesn't do anything special. It's a branch.

Adds branch deletion functionality.
- Branches can be removed by pressing `LT` (or `G` when using the keyboard).
- Removed (and broken) branches spawn an ugly little stumpy knot where they once were, preventing new branches from being placed there.

Yes, I'm sorry, I added rigid body and secondary collider to the Squirrel.
- I tried doing it several other ways but every time I opened Visual Studio a portal to the elemental plane of code spaghetti opened and it made me quite annoyed
- The rigidbody is no-gravity, kinematic only. That is to say, it doesn't apply physics to the Squirrel in any way whatsoever.
- The collider exists to fire collision events on the branches when the Squirrel lands on it.

**BIG HUGE DISCLAIMER:**
As a general continuation of my commentary on #63, the squirrel player movement implementation remains difficult to work around. As it currently exists in `master`, it has no notion of acceleration or deceleration beyond the manual gravity implementation. The specific issue is that `_moveDirection` is treated _only_ as instantaneous velocity. Implementing "bouncing" and "sliding" _requires_ some sort of change in velocity _over time_. To get around this, I implemented a `_velocity` field that applies itself to `_moveDirection` every tick and then decays slightly. This doesn't work very well but improvements will require a rework of `PlayerController` internals - unless somebody has an idea as to how to implement acceleration/deceleration without it.

Yes, this means that the Bouncy and Slippery branches don't always work properly but I wanted to get this PR up before another week went by, maybe somebody else has some ideas as to how to approach this.

As an alternative to accepting this PR, work can continue on this branch until a better solution is found.

NicolaeChristof commented:
Just an update, I still havent had time to review these additions, I will try to get to them later today or tomorrow at the latest.

malcolmriley commented:
Changes since last review:
- UglyKnot collider is now trigger type.
- Removed "bad" slipping and bouncing.

I can't replicate your issue with the branch deletion. It is working on my end.

To "fix" the slipping and bouncing: Do I have permission to do a rewrite of `PlayerController`?

malcolmriley commented:
Merge conflicts should be resolved. Please test.
